### DISCUSSION TITLE
Subscription

### DISCUSSION THREAD

**Discussion by bentrynning** [medusajs/medusa#2942] [Feature Requests]
Basic core subscription service and payment service to be used in storefront and admin. 

The ide of having a basic starting point of subscription feature that can be extended. 
It could be built from scratch today, but for beginners and just to get something up and running fast it would be nice to have this as part of the core when staring 👍

**Comment by srindom** [comment]
Thanks! Could you try adding more details to what a "basic starting point" would include? :)

**Comment by pevey** [comment]
I'll take a stab at listing some requirements:

- Ability to configure a product as either one-time-purchase (default) or subscription
- If cart includes a subscription item, create a subscription with payment provider instead of one-time payment intent.  (May have to require cart includes only a sub OR a one-time-payment, but not both.  It would be great if cart could include both and somehow Medusa could create both payment flows.  I don't know if that is possible.  Will need looking into.
- Recurring payments handled by medusa backend
- Authenticated store endpoint for customer to cancel subscription
- Authenticated store endpoint for customer to see subscription status, including recent payments
- Authenticated admin endpoints for managing customer subscriptions.  1) Cancel sub and notify customer, 2) retry payment 3) Get status, and recent activity
- Configurable integration with fulfillment.  With each subscription payment, is new fulfillment required (e.g., shipping a physical product?

I'm sure there is more I'm not thinking of at the moment.

**Comment by rickyshowtime** [comment]
This is a really anticipated feature for our team !

We are currently developping a B2B store with a subscription model (plus a whole lot of other features like distance based shipping cost calculation etc) but this is a must have for us that we will need to develop on our side if not planned by Medusa. 

We would love to contribute as soon as this get started !

**Reply by willywongi** [reply]
Take a look at what [Lago](https://www.getlago.com/) is doing. I think they already solved many of the problems of subscriptions or usage based services.

**Comment by chrismarut** [comment]
We've been working on solving the hard parts of recurring commerce and subscription features inside Medusa: renewals, failed payment recovery (dunning), cancellation & retention workflows, and subscription analytics. All modeled as clean, separated domains within a single plugin architecture

We just published a full technical demo walking through the entire system — real admin workflows, real customer scenarios, and the architectural decisions behind them:
https://www.youtube.com/watch?v=KY43_6Q3560&t=2s

If you're curious about what's under the hood:
📦 npm: https://www.npmjs.com/package/@reorderjs/reorder
💻 GitHub: https://github.com/reorder-js
🌐 Website: https://reorderjs.com/

The repo is public and the plugin is installable via npm, so feel free to dig in, use it as a starting point, or just study the patterns if you're building anything subscription-related on Medusa.
We'd love to hear your thoughts — and if you have any questions about the implementation
